granule_metadata_extractor/processing/process_scrxsondecpexaw.py

```diff
@@ -0,0 +1,139 @@
+from ..src.extract_netcdf_metadata import ExtractNetCDFMetadata
+import os
+import numpy as np
+from netCDF4 import Dataset
+from datetime import datetime, timedelta
+
+class ExtractScrxsondecpexawMetadata(ExtractNetCDFMetadata):
+    """
+    A class to extract scrxsondecpexaw
+    """
+
+    def __init__(self, file_path):
+        self.file_path = file_path
+        # these are needed to metadata extractor
+        if file_path.endswith('.nc'):
+           self.fileformat = 'netCDF-4'
+
+           # extracting time and space metadata for netcdf/ascii file
+           [self.minTime, self.maxTime, self.SLat, self.NLat, self.WLon, self.ELon] = \
+               self.get_variables_min_max_netcdf()
+        else: #ascii (.txt)
+           self.fileformat = 'ASCII'
+
+           # extracting time and space metadata for netcdf/ascii file
+           [self.minTime, self.maxTime, self.SLat, self.NLat, self.WLon, self.ELon] = \
+               self.get_variables_min_max_ascii()
+
+    def get_variables_min_max_netcdf(self):
+        """
+        :return:
+        """
+        data = Dataset(self.file_path)
+        ref_time = datetime.strptime(data.start_date_and_time,'%Y/%m/%d %H:%M:%S') #i.e.,'2021/08/19 23:12:00'
+        sec = data['STANDARDVARS/TIME'][:] #seconds since start_date_and_time
+        lon = data['STANDARDVARS/LONG'][:] #degreeE
+        lat = data['STANDARDVARS/LATI'][:] #degreeN
+        data.close()
+
+        minTime = ref_time + timedelta(seconds = min(sec))
+        maxTime = ref_time + timedelta(seconds = max(sec))
+        maxlat, minlat, maxlon, minlon = [max(lat),
+                                          min(lat),
+                                          max(lon),
+                                          min(lon)]
+
+        return minTime, maxTime, minlat, maxlat, minlon, maxlon
+
+    def get_variables_min_max_ascii(self):
+        """
+        :return:
+        """
+
+        fname = self.file_path.split('/')[-1]
+        if '_unix_' in fname:
+           #Sample file SCRX_Radiosonde_CPEXAW_unix_20210830_1301.txt
+           key = fname.split('.')[0].split('_CPEXAW_unix_')[-1] #i.e.,20210819_2312
+           with open(self.file_path,'r') as f:
+                lines = f.readlines()
+        else: #'_win_' in fname
+           key = fname.split('.')[0].split('_CPEXAW_win_')[-1] #i.e.,20210819_2312
+           with open(self.file_path,'rb') as f:
+                lines0 = f.readlines()
+           lines = [x.decode('utf-8') for x in lines0[1:]]

```

Comment:
You should be able to decode in the open command with something like
```
with open(self.file_path, 'rb', 'utf-8') as f:
```
that way you don't have to read in the file then loop through it again to decode

Reply:
I see now that your current implementation is to get rid of an unusual header line which was causing issues. Your implementation is best.